sdk/tests/iot/hub/test_az_iot_hub_client.c

```diff
@@ -122,7 +122,7 @@ static void test_az_iot_hub_client_get_client_id_NULL_client_fails(void** state)
       az_iot_hub_client_get_client_id(NULL, test_buf, sizeof(test_buf), &test_length));
 }
 
-static void test_az_iot_hub_client_get_client_id_NULL_input_span_fails(void** state)
+static void test_az_iot_hub_client_get_client_id_NULL_char_buf_fails(void** state)
 {
```

Comment:
The only reason I would expand buf to buffer in the naming (here and elsewhere) is because of this note in the C guidelines: https://azure.github.io/azure-sdk/clang_design.html#clang-design-naming-concise

But I'm not too concerned about that in the test file.  Your choice :)